Reference/google/protobuf/compiler/plugin.pb.swift

```diff
@@ -79,7 +79,17 @@ public struct Google_Protobuf_Compiler_CodeGeneratorRequest: ProtobufGeneratedMe
   public var fileToGenerate: [String] = []
 
   ///   The generator parameter passed on the command-line.
-  public var parameter: String? = nil
+  private var _parameter: String? = nil
+  public var parameter: String {
+    get {return _parameter ?? ""}
+    set {_parameter = newValue}
+  }
+  public var hasParameter: Bool {
+    return _parameter != nil
+  }
+  public mutating func clearParameter() {
+    return _parameter = nil
+  }
```

Comment:
> The large number of force-unwraps that I removed in the generator code after I updated descriptor.pb.swift to use non-optional properties is a sign to me that the user-facing API was not ideal to work with.

Wow.  Your "large number" is a lot more than I expected.   After playing with 'grep' a bit, it looks like there are really very few force-unwraps left in the generator codebase, and it looks like it would not take much work to reduce it further.